app/src/main/java/com/example/greetingcard/MainActivity.kt

```diff
@@ -22,31 +31,73 @@ class MainActivity : ComponentActivity() {
         enableEdgeToEdge()
         setContent {
             GreetingCardTheme {
-                Scaffold(modifier = Modifier.fillMaxSize()) { innerPadding ->
-                    Greeting(
-                        name = "Android",
-                        modifier = Modifier.padding(innerPadding)
-                    )
-                }
+                LoginScreen()
             }
         }
     }
 }
 
 @Composable
-fun Greeting(name: String, modifier: Modifier = Modifier) {
-    Surface(color = Color.Cyan) {
-        Text(
-            text = "Hi, my name is $name!",
-            modifier = modifier.padding(24.dp)
+fun LoginScreen() {
+
+    val emailState = remember { mutableStateOf("") }
+    val passwordState = remember { mutableStateOf("") }
+
+    Scaffold { innerPadding ->
+        Column(
+            modifier = Modifier
+                .padding(innerPadding)
+                .padding(32.dp),
+        ) {
+
+            Spacer(modifier = Modifier.height(24.dp))
+
+            Title()
+
+            Spacer(modifier = Modifier.height(48.dp))
+
+            InputFields(emailState, "E-mail", KeyboardType.Email)
+
+            Spacer(modifier = Modifier.height(36.dp))
+
+            InputFields(passwordState, "Senha", KeyboardType.Password)
+
+            Spacer(modifier = Modifier.height(36.dp))
+
+            LoginButton()
+        }
+    }
+}
+
+@Composable
+fun Title() {
+    Text(
+        fontWeight = FontWeight.Bold, fontSize = 24.sp, text = "Bem vindo(a) à Taqtile!"
+    )
+}
+
+@Composable
+fun InputFields(state: androidx.compose.runtime.MutableState<String>, title: String, type: KeyboardType) {
```

Comment:
Aqui tem uma diferença sutil, mas que é uma melhor prática: sabe o state que você está passando aqui? Ele pertence ao composable `LoginScreen`, certo? Então não é muito legal você passar o estado como um todo pra outros componentes e deixar que esses componentes o modifiquem.

Por uma questão de SOLID (organização, manter os papéis nos devidos escopos), é melhor que você passe para os outros componentes, caso eles precisem, 2 coisas:

1. O valor do estado apenas. Neste caso, seria a string com o valor do email
2. Uma função para "o que fazer quando a textfield mudar de valor". Assim você implementa uma função lá no `LoginScreen` (que vai ser triggada toda a vez o valor do campo trocar) para poder realizar a troca de estado no lugar correto.

Algo assim:

```kotlin
val emailState = remember { mutableStateOf("") }

fun changeEmailState(email: String) {
  emailState.value = email
}

// ...
InputFields(onValueChange = { email -> changeEmailState(email) })

// ...
@Composable
fun InputFields(title: String, type: KeyboardType, onValueChange: (String) -> Void) {
```

Ficou claro?